src/client-side-metrics/operation-metrics-collector.ts

```diff
@@ -76,6 +76,8 @@ export class OperationMetricsCollector {
   private serverTime: number | null;
   private connectivityErrorCount: number;
   private streamingOperation: StreamingState;
+  private applicationLatencies: number[];
+  private lastRowReceivedTime: Date | null;
```

Comment:
is Date a safe way to store timestamps in node? 

My understanding is that Date is based on the system time, so it's not monotonic, and could move backwards. A quick google search shows [process.hrtime](https://nodejs.org/api/process.html#processhrtimebigint) might be a better fit for this kind of thing

Reply:
We only measure deltas between dates, but just to be safe I switched to using hrtime everywhere.